subsys/net/l2/wifi/wifi_shell.c

```diff
@@ -501,6 +501,7 @@ static int __wifi_args_to_params(const struct shell *sh, size_t argc, char *argv
 				params->psk_length = strlen(params->psk);
 			} else {
 				PR_WARNING("Passphrase provided without security configuration\n");
+				return -EINVAL;
```

Comment:
This puts a dependency on the order, e.g., if `-p X -k Y` it would fail with `Passphrase provided without security configuration`